### PR TITLE
Added support for acl authentication.

### LINKED ISSUE #30: Support databases protected with ACL (user + password authentication)
gears_cli allows providing password as part of authentication to the database, however, from Redis 6 databases can be protected by both username and password.
It is needed to allow gears_cli operations with ACL defined DBs.

### PR DESCRIPTION
**Pull request by MeirShpilraien**
Fix #30, added support for acl authentication.

### CODE REVIEW

File: pytests/example.py [hunk 1]
```diff
@@ -0,0 +1 @@
+GB('ShardsIDReader').run()
```
[the remark this answers is not in the document]
Comment (gkorland) [reply pytests/example.py:1]: ignore

File: pyproject.toml [hunk 2]
```diff
@@ -20,7 +20,7 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = "^3.6"
-redis = "3.5.3"
+redis = "4.1.3"
```
Comment (chayim) [inline pyproject.toml]:
```
redis = "^4.1.3"
```

This keeps it within the same minor patch set - and we've already released 4.1.4

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[tox]
+skipsdist = true
+envlist = tests
+
+# These are the tests that run in CI. They can be run locally in a redislabs/redisgears:edge docker
+[testenv:tests]
+commands_pre =
+    pip install --upgrade pip
+    {envdir}/bin/python -m pip install poetry
+    {envdir}/bin/python -m pip install RLTest==0.5
```
Comment (chayim) [inline tox.ini]:
```
"RLTest~=0.5.0"
```
Isn't this  to keep within the same patch set?

File: pyproject.toml (hunk 2 above)
Comment (MeirShpilraien) [reply pyproject.toml]: Fixed

File: tox.ini (hunk 3 above)
Comment (MeirShpilraien) [reply tox.ini]: Fixed

**Pull request merged**